test/functional/testlib.bash

```diff
@@ -1780,22 +1778,47 @@ destroy_test_fs() { # swupd_function
 
 create_config_file() { # swupd_function
 
+	SWUPD_CONFIG_DIR="$("$SWUPD" --version | grep "config file path" | awk '{print $4}')"
+
+	# when the config dir is set to a relative path, lets redirect it to the
+	# directory of the test environment so every test has its own config file
+	# this is useful when running the config tests in parallel with other tests,
+	# if the path is absolute lets leave it as it is
+	if [ "$SWUPD_CONFIG_DIR" == "./testconfig" ]; then
```

Comment:
Why do we need to have this case supported? And why it should be specific for ./testconfig/

Reply:
well, currently if we want to run the tests in parallel, we need to create one config file per test, which forces us to change the directory where swupd reads the config file from. So if the user wants to run tests in parallel he/she will need to configure the build with ` --with-config-file-path=./testconfig` like we are doing right now. That is why I left this particular condition here, I don't see how else we could support multiple config files without having to add a flag for setting up the config file. I'm open to suggestions if you have any.